### PR TITLE
Allow setting elliptic curves using ssl_curve_list parameter

### PR DESCRIPTION
**Pull request by juhakoho**
Hello,

Please consider this PR that will add the possibility to set the elliptic curves using a new parameter "ssl_curve_list".

### CODE REVIEW

**Comment by sirainen** [comment]
This would at least need some #ifdef to build with older OpenSSL versions. Also at the same time it should be added to lib-ssl-iostream (the ssl-proxy-openssl.c code duplication is going away in v2.3).

**Comment by juhakoho** [comment]
Thanks for clearing this up. I was a bit confused about lib-ssl-iostream and duplication in ssl-proxy-openssl.

The new commit I suppose fixes the issues you mentioned.

File: src/lib-ssl-iostream/iostream-openssl.c [hunk 1]
```diff
@@ -160,6 +160,17 @@ openssl_iostream_set(struct ssl_iostream *ssl_io,
 			return -1;
 		}
 	}
+#if OPENSSL_VERSION_NUMBER >= 0x10002000L
+	if (set->curve_list != NULL && strlen(set->curve_list) > 0 &&
+		strcmp(ctx_set->curve_list, set->curve_list) != 0) {
```
Comment (sirainen) [inline src/lib-ssl-iostream/iostream-openssl.c]: It could be that set->curve_list != NULL, but ctx_set->curve_list == NULL and this would crash then.

Comment (juhakoho) [inline src/lib-ssl-iostream/iostream-openssl.c]: Good point... fixed. At least I assume that ctx_set->curve_list being NULL should behave like if the strings are not equal.

**Comment by cmouse** [comment]
hi! can you rebase your work?

**Comment by juhakoho** [comment]
Hi, sorry for the late response but I was on holidays. Should be okay now.

**Comment by mrannanj** [comment]
A slightly modified version of this is in internal review.

**Comment by juhakoho** [comment]
Great! Thanks!

**Comment by sirainen** [comment]
Merged

**Pull request closed without merging**